wrap/src/components/WrapForm/index.tsx

```diff
@@ -79,7 +79,14 @@ export const WrapForm: FC = () => {
 
       navigate(`/wrap/tx/${txHash}?amount=${value}&action=${formType}`)
     } catch (ex) {
-      setError((ex as Error)?.message || JSON.stringify(ex))
+      if (
+        typeof ex === 'object' &&
+        (ex as { details: string }).details === 'MetaMask Tx Signature: User denied transaction signature.'
```

Comment:
This is bad and only works in metamask

Looks like viem wraps the eip-1193 error deep in `ex.cause.cause.code === 4001`. Can also be detected by `ex.cause.cause instanceof UserRejectedRequestError`. Based on https://viem.sh/docs/contract/simulateContract#handling-custom-errors maybe `ex.walk(err => err instanceof UserRejectedRequestError)`.

But let's just use `ex.shortMessage: 'User rejected the request.'`